src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -140,7 +142,15 @@ public double getHappiness()
                 totalWeight += happinessModifier.getWeight();
             }
 
-            cachedHappiness = Math.min(10.0 * (total / totalWeight), 10);
+            double happinessResult = (total / totalWeight);
+            final MultiplierModifierResearchEffect xpBonus =
+              colony.getResearchManager().getResearchEffects().getEffect(LEVELING, MultiplierModifierResearchEffect.class);
```

Comment:
leveling bonus for xp applied to happiness?

Reply:
yee, fixed